Update fsharp tree sitter repo reference

ssiltanen opened the pull request:
The repository reference used here was a fork of the actual repository https://github.com/Nsidorenco/tree-sitter-fsharp, which has now been moved under ionide organization https://github.com/ionide/tree-sitter-fsharp, where it is in active maintenance and development.

The commit SHA is the currently latest commit from main branch.

The injections.scm is copied as is from the fsharp treesitter repo [queries](https://github.com/ionide/tree-sitter-fsharp/blame/main/queries).

The locals.scm is copied from the repo and the capture names are updated to follow the standard names:

    - Replace @local.definition.var @local.definition.function, and @local.definition.parameter with @local.definition
    - Remove (#set! "definition.function.scope" "parent")

The highlights.scm is copied as well from the fsharp treesitter repo, but modified here to match helix highlight scopes based on my best guesstimates. The changes made based on syntax highlight documentation [here](https://docs.helix-editor.com/master/themes.html#syntax-highlighting):
    
    - Remove @spell scopes
    - Split @comment into @comment.line and @comment.block
    - Replace @comment.documentation with @comment.block.documentation
    - Replace @character.special with @special
    - Replace @variable.member with @variable.other.member
    - Replace @type.definition with @type
    - Replace @function.member with @function.method
    - Replace @module with @namespace
    - Replace @constant.macro with @function.macro
    - Replace @property with @variable.other.member
    - Replace @variable.member with @variable.other.member
    - Replace @variable.parameter.builtin with @variable.builtin
    - Replace @function.call with @function
    - Replace @number with @constant.numeric.integer and @constant.numeric.float
    - Replace @boolean with @constant.builtin.boolean
    - Replace @keyword.conditional with @keyword.control.conditional
    - Replace @keyword.return with @keyword.control.return
    - Replace @keyword.repeate with @keyword.control.repeat
    - Replace @keyword.import with @keyword.control.import
    - Replace @keyword.modifier with @keyword.storage.modifier
    - Replace @keyword.type with @keyword.storage.type
    - Replace @keyword.exception with @keyword.control.exception
    - Replace @module.builtin with @namespace

the-mikedavis commented:
When the revision is updated the queries (`runtime/queries/fsharp/*.scm`) usually need to be updated as well. The docs CI is failing because the old queries are no longer valid for the new version of the grammar

ssiltanen commented:
Thanks for the tip! I ended up copying the queries files from the fsharp treesitter repo. Hopefully that was what you meant, at least the build now succeeded. I don't really have the know-how to validate the highlights changes further.

the-mikedavis commented:
The queries in the repo are for neovim it looks like. `injections.scm` is ok but the others should be reverted. For `highlights.scm` we use different scope names: https://docs.helix-editor.com/master/themes.html#syntax-highlighting

ssiltanen commented:
The highlights are now updated to match the helix scope keywords. Mainly I think these should be fine, but I did not find a suitable candidate for the `@spell` or `@character.special`. The latter I ended up switching to `@special`, however the documentation did not provide much info on its usage, so it could be completely wrong. I also considered `@string.special`.

ssiltanen commented:
Also, just making sure that did you mean to revert the `locals.scm` as well instead of using the updated version from the fsharp treesitter repo?

the-mikedavis commented:
Yeah the locals.scm is in neovim format. We follow tree-sitter so you could try to coerce that locals.scm upstream into the captures we use: `local.definition`, `local.scope` and `local.reference` (just those three exactly, we don't allow the trailing stuff like `local.definition.parameter`) https://tree-sitter.github.io/tree-sitter/syntax-highlighting#local-variables

the-mikedavis commented:
`@spell` is for spellchecking in neovim. We don't have that yet and we won't add that to `highlights.scm` so that one is ok to leave out. `@special` for `@character.special` seems ok to me

ssiltanen commented:
I have now updated the locals.scm based on the fsharp tree-sitter locals, and replaced the capture names to @local.scope, @local.definition and @local.reference. I was unsure how to deal with the `#set!` , but if I have understood it correctly it is neovim specific thing, which is why I removed it.

ssiltanen commented on runtime/queries/fsharp/locals.scm:
```diff
@@ -1,25 +1,33 @@
+(function_declaration_left
+  .
+  ((_) @local.definition
+    (#set! "definition.function.scope" "parent"))
```
Removed the `(#set! "definition.function.scope" "parent")` but left the surrounding `((_) @local.definition)`
